src/app/api/booking/route.ts

```diff
@@ -0,0 +1,117 @@
+    if (currentUser === null)
+      return new Response("unauthorized user", { status: 401 });
+
+    const json = await request.json();
+    const num_of_sessions = json.length;
+
+    if (num_of_sessions === 0)
+      return new Response("num_of_sessions cannot be equal to 0", {
+        status: 400,
+      });
+
+    const user = await db.query.users.findFirst({
+      where: eq(users.id, currentUser!.id),
+    });
+
+    if (user?.member === true) {
+      if (num_of_sessions > 2) {
+        return new Response("num_of_sessions cannot be greater than 2", {
+          status: 400,
+        });
+      } else {
+        for (const session of json) {
+          const gameSession = await db.query.gameSessions.findFirst({
+            where: eq(gameSessions.id, session.gamesessionId),
+          });
+
+          const time = new Date();
+          if (time > gameSession!.endTime || time < gameSession!.startTime) {
+            return new Response(null, { status: 400 });
+          }
+        }
+        if (user.remainingSessions < num_of_sessions) {
+          return new Response(null, { status: 400 });
+        }
+
+        await db.transaction(async (tx) => {
+          await tx
+            .update(users)
+            .set({
+              remainingSessions: user.remainingSessions - num_of_sessions,
+            })
+            .where(eq(users.id, currentUser!.id));
+          await tx.insert(bookings).values({
+            userId: currentUser!.id,
+            gameSessionId: json[0].gamesessionId,
+            createdAt: new Date(),
+            difficulty: json[0].playLevel,
+          });
+          await tx.insert(bookings).values({
+            userId: currentUser!.id,
+            gameSessionId: json[1].gamesessionId,
+            createdAt: new Date(),
+            difficulty: json[1].playLevel,
+          });
+        });
+
+        return new Response("success", { status: 204 });
+      }
+    } else {
+      if (num_of_sessions !== 1) {
+        return new Response("num_of_sessions is not equal to 1", {
+          status: 400,
+        });
+      } else {
+        const gameSession = await db.query.gameSessions.findFirst({
+          where: eq(gameSessions.id, json[0].gamesessionId),
+        });
+
+        const time = new Date();
+        if (time > gameSession!.endTime || time < gameSession!.startTime) {
+          return new Response(null, { status: 400 });
+        }
+        if (user!.remainingSessions < num_of_sessions) {
+          return new Response(null, { status: 400 });
+        }
+
+        await db.transaction(async (tx) => {
+          await tx
+            .update(users)
+            .set({
+              remainingSessions: user!.remainingSessions - num_of_sessions,
+            })
+            .where(eq(users.id, currentUser!.id));
+          await tx.insert(bookings).values({
+            userId: currentUser!.id,
+            gameSessionId: json[0].gamesessionId,
+            createdAt: new Date(),
```

Comment:
has a default value assigned on the database level (defined in the schema) so this line is ok to omit

Reply:
should u change the ticket then? cuz then there is no need for difficulty input right?

Reply:
oh it didnt highlight the line correctly, i meant the `createdAt` field sorry

Reply:
my man! yeah i just realised that oops